src/k8s/pkg/k8sd/api/utils/k8s-dqlite.go

```diff
@@ -205,20 +150,27 @@ func waitForNodeJoin(ctx context.Context, host string) error {
 	ch := make(chan struct{}, 1)
 	go func() {
 		for {
-			cmd := exec.Command(
-				snap.Path("bin/dqlite"),
-				"-s", fmt.Sprintf("file://%s/cluster.yaml", clusterDir),
-				"-c", fmt.Sprintf("%s/cluster.crt", clusterDir),
-				"-k", fmt.Sprintf("%s/cluster.key", clusterDir),
-				"-f", "json", "k8s", ".cluster",
-			)
-
-			out, err := cmd.CombinedOutput()
-			if err == nil && strings.Contains(string(out), host) {
-				break
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				// TODO: Use go-dqlite lib instead of shelling out.
+				cmd := exec.Command(
+					snap.Path("bin/dqlite"),
+					"-s", fmt.Sprintf("file://%s/cluster.yaml", clusterDir),
+					"-c", fmt.Sprintf("%s/cluster.crt", clusterDir),
+					"-k", fmt.Sprintf("%s/cluster.key", clusterDir),
+					"-f", "json", "k8s", ".cluster",
+				)
+
+				out, err := cmd.CombinedOutput()
+				if err == nil && strings.Contains(string(out), host) {
+					ch <- struct{}{}
+					return
+				}
 			}
+			time.Sleep(time.Second)
```

Comment:
I feel like this makes this less readable.
If I want to check how long the `waitForNodeJoin` function wait, I'd need to check the top of the file. Let's keep this close to the implementation since we don't expect the user to do anything with the timeouts anyway.